idrstream/merge_CP_DP.py

```diff
@@ -0,0 +1,162 @@
+"""
+This file holds the functions necessary for merging IDR_stream CP and DP outputs.
+As the IDR_stream output follows the pycytominer output format, these merge functions should work for any pycytominer output data.
+"""
+
+import pandas as pd
+import pathlib
+import math
+
+
+def merge_CP_DP_batch_data(
+    cp_batch_data: pd.DataFrame, dp_batch_data: pd.DataFrame
+) -> pd.DataFrame:
+    """
+    merge dataframes for IDR_stream output with CP and DP features
+    the two features dataframes should have aligned location metadata (plate, well, frame, etc) and the same number of rows (cells)
+
+    Parameters
+    ----------
+    cp_batch_data : pd.DataFrame
+        idrstream_cp batch output
+    dp_batch_data : pd.DataFrame
+        idrstream_dp batch output
+
+    Returns
+    -------
+    pd.DataFrame
+        merged batch data with metadata, CP features, and DP features
+
+    Raises
+    ------
+    IndexError
+        cp and dp dataframes have different number of rows (cells)
+    """
+
+    # check batch data have same number of rows (cells)
+    # if batch data have different number of cells, raise an error because they must not have close segmentations
+    if cp_batch_data.shape[0] != dp_batch_data.shape[0]:
+        raise IndexError("Batch data have different number of rows (cells)!")
+
+    # hide warning for pandas chained assignment
+    pd.options.mode.chained_assignment = None
+
+    # get cp and dp column names
+    cp_columns = cp_batch_data.columns
+    dp_columns = dp_batch_data.columns
+    # get metadata columns (columns that show up in both dataframes)
+    metadata_columns = [col for col in cp_columns if col in dp_columns]
```

Comment:
Would there ever be metadata columns which have different names between `cp_columns` and `dp_columns` but should still be excluded from results?

Reply:
Possibly... but I think trying to include those here goes a bit out of the scope of this PR.

Reply:
Thanks - it could be that another approach avoiding self-inference could improve things, for example if all the columns include a specific string or follow another similar pattern. I'd defer to you on what's best with the given datasets involved.